I hope this helps - the issue only happens with pyhon2, maybe the cache does not support it fully yet?
```py
git log -n1 --oneline (with no changes in the tree)
6eb942cf45 (HEAD) Speedup environment activation, part 2 (#25633)
spack -d build-env chill
...
==> ...'/opt/spack/linux-ubuntu18.04-zen/gcc-8.4.0/python-2.7.18-oq7rprqjddoe5kovw2tvwvlwt3hizfkx/bin/python2.7' '-c' '
import json
from distutils.sysconfig import (
    get_config_vars,
    get_config_h_filename,
    get_makefile_filename,
    get_python_inc,
    get_python_lib,
)

config = get_config_vars()
config['"'"'config_h_filename'"'"'] = get_config_h_filename()
config['"'"'makefile_filename'"'"'] = get_makefile_filename()
config['"'"'python_inc'"'"'] = {}
config['"'"'python_lib'"'"'] = {}

for plat_specific in [True, False]:
    config['"'"'python_inc'"'"'][plat_specific] = get_python_inc(plat_specific, prefix='"'"''"'"')
    config['"'"'python_lib'"'"'][plat_specific] = {}
    for standard_lib in [True, False]:
        config['"'"'python_lib'"'"'][plat_specific][standard_lib] = get_python_lib(
            plat_specific, standard_lib, prefix='"'"''"'"'
        )

print json.dumps(config)
'
Traceback (most recent call last):
  File "/home/kaindlb/spack/spack-stable-Barney/bin/spack", line 100, in <module>
    sys.exit(spack.main.main())
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/main.py", line 797, in main
    return _invoke_command(command, parser, args, unknown)
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/main.py", line 525, in _invoke_command
    return_val = command(parser, args)
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/cmd/build_env.py", line 16, in build_env
    env_utility.emulate_env_utility('build-env', 'build', args)
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/cmd/common/env_utility.py", line 67, in emulate_env_utility
    build_environment.setup_package(spec.package, args.dirty, context)
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/build_environment.py", line 771, in setup_package
    pkg.spec, context, custom_mods_only=False))
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/build_environment.py", line 964, in modifications_from_dependencies
    add_modifications_for_dep(dspec)
  File "/home/kaindlb/spack/spack-stable-Barney/lib/spack/spack/build_environment.py", line 945, in add_modifications_for_dep
    dpkg.setup_dependent_package(spec.package.module, spec)
  File "/home/kaindlb/spack/spack-stable-Barney/var/spack/repos/builtin/packages/python/package.py", line 1003, in setup_dependent_package
    module.python_include_dir = join_path(dependent_spec.prefix,
  File "/home/kaindlb/spack/spack-stable-Barney/var/spack/repos/builtin/packages/python/package.py", line 870, in python_lib_dir
    else:
KeyError: 'false'
```
I saved the command to a file, and piped the output to json_pp, the output looks fine:
```json
bash saved-script|json_pp|grep LIBDIR
   "LIBDIR" : "/home/kaindlb/spack/spack-stable-Barney/opt/spack/linux-ubuntu18.04-zen/gcc-8.4.0/python-2.7.18-oq7rprqjddoe5kovw2tvwvlwt3hizfkx/lib",
```

--- a/var/spack/repos/builtin/packages/python/package.py
+++ b/var/spack/repos/builtin/packages/python/package.py
@@ -717,10 +717,12 @@ config['python_inc'] = {}
 config['python_lib'] = {}
 
 for plat_specific in [True, False]:
-    config['python_inc'][plat_specific] = get_python_inc(plat_specific, prefix='')
-    config['python_lib'][plat_specific] = {}
+    plat_key = str(plat_specific).lower()
+    config['python_inc'][plat_key] = get_python_inc(plat_specific, prefix='')
+    config['python_lib'][plat_key] = {}
     for standard_lib in [True, False]:
-        config['python_lib'][plat_specific][standard_lib] = get_python_lib(
+        lib_key = str(standard_lib).lower()
+        config['python_lib'][plat_key][lib_key] = get_python_lib(
             plat_specific, standard_lib, prefix=''
         )
 
@@ -841,9 +843,9 @@ for plat_specific in [True, False]:
         Returns:
             str: include files directory
         """
-        if 'python_inc' in self.config_vars:
+        try:
             return self.config_vars['python_inc']['false']
-        else:
+        except KeyError:
             return os.path.join('include', 'python{0}'.format(self.version.up_to(2)))
 
     @property
@@ -865,9 +867,9 @@ for plat_specific in [True, False]:
         Returns:
             str: standard library directory
         """
-        if 'python_lib' in self.config_vars:
+        try:
             return self.config_vars['python_lib']['false']['true']
-        else:
+        except KeyError:
             return os.path.join('lib', 'python{0}'.format(self.version.up_to(2)))
 
     @property
@@ -889,9 +891,9 @@ for plat_specific in [True, False]:
         Returns:
             str: site-packages directory
         """
-        if 'python_lib' in self.config_vars:
+        try:
             return self.config_vars['python_lib']['false']['false']
-        else:
+        except KeyError:
             return self.default_site_packages_dir
 
     @property

